### PR TITLE
Merging feature branch git CG k8s secret support

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Merging feature branch git CG k8s secret support

Screenshot 2023-10-10 at 1 37 33 PM

------------------------------------------------

Screenshot 2023-10-26 at 11 44 40 AM

### CODE REVIEW

**Review by ishangupta-ds — DISMISSED** [review]
@Shubbhang351 please attach a screenshot or link of a successful acceptance test pipeline run before the merge. Cc: @ramya-bangera

**Review by shobha2626 — APPROVED** [review]
LGTM

**Comment by ishangupta-ds** [comment]
@Shubbhang351 the test failures are expected?

File: docs/data-sources/kubernetes_secret.md [hunk 1]
```diff
@@ -10,20 +10,21 @@ Manage a Kubernetes Secret using this Terraform module.
 
 In managed clusters, both attached and provisioned, you can create Kubernetes Secret that you can manage through Tanzu Mission Control.
 
-To create a cluster secret, you must be associated with the cluster.admin role
+To create a kubernates secret, you must be associated with the cluster.admin or clustergroup.admin role.
```
Comment (ramya-bangera) [inline docs/data-sources/kubernetes_secret.md:13]: Fix this typo here and throughout the docs

Comment (Shubbhang351) [reply docs/data-sources/kubernetes_secret.md]: Updated

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**